x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingResourceTracker.java

```diff
@@ -208,47 +219,118 @@ static void getMemoryAndProcessors(
+
+                if (numberOfAvailableProcessors < numProcessorsNeeded || perNodeAvailableModelMemoryBytes < numMemoryBytesNeeded) {
+                    // we don't have enough processors or memory to start the new allocations, we need to scale up
+
+                    processorsNewlyAssignedToTrainedModelsAllocationsThisTime += numExistingProcessorsToBeUsed;
+
+                    extraProcessors += numMissingAllocations * numberOfThreadsPerAllocation - numExistingProcessorsToBeUsed;
+                    extraSingleNodeProcessors = Math.max(extraSingleNodeProcessors, numberOfThreadsPerAllocation);
+                    extraModelMemoryInBytes += numMemoryBytesNeeded;
+                    extraPerNodeModelMemoryBytes = Math.max(extraPerNodeModelMemoryBytes, numMemoryBytesNeeded);
+                    final int extraProcessorsNeededForAssignment = numMissingAllocations * numberOfThreadsPerAllocation
+                        - numExistingProcessorsToBeUsed;
+                    logger.warn(
+                        () -> format(
+                            "trained model [%s] assigned to [%s], waiting for [%d] allocations to start; waiting on [%d] extra processors "
+                                + "to become available",
+                            modelAssignment.getKey(),
+                            Strings.arrayToCommaDelimitedString(modelAssignment.getValue().getStartedNodes()),
+                            numMissingAllocations,
+                            extraProcessorsNeededForAssignment
+                        )
+                    );
+                } else {
+                    logger.info(
+                        () -> format(
+                            "trained model [%s] assigned to [%s], waiting for [%d] allocations to start",
+                            modelAssignment.getKey(),
+                            Strings.arrayToCommaDelimitedString(modelAssignment.getValue().getStartedNodes()),
+                            numMissingAllocations
+                        )
+                    );
+                    processorsNewlyAssignedToTrainedModelsAllocationsThisTime += numExistingProcessorsToBeUsed;
+                }
+
+                existingModelMemoryBytes += estimatedMemoryUsage;
+                sumOfCurrentlyExistingAndUsedProcessors += numTargetAllocationsOnExistingNodes * numberOfThreadsPerAllocation;
+
+                for (String node : modelAssignment.getValue().getNodeRoutingTable().keySet()) {
+                    perNodeModelMemoryInBytes.computeIfAbsent(node, k -> new ArrayList<>())
+                        .add(
+                            MlJobRequirements.of(
+                                estimatedMemoryUsage,
+                                Priority.LOW.equals(modelAssignment.getValue().getTaskParams().getPriority())
+                                    ? 0
+                                    : numberOfThreadsPerAllocation
```

Comment:
seems like yes?

Reply:
ah, down in `checkIfOneNodeCouldBeRemoved` where this is used, there is this comment         "we define least loaded exclusively on memory, not CPU load" so I think it doesn't matter in this context. I think I am going to make it zero so that we have to fix it later if we do scale down by processors.